cmp/viewmanager/ViewManagerModel.ts

```diff
@@ -471,31 +469,45 @@ export class ViewManagerModel<T = PlainObject> extends HoistModel {
     // Implementation
     //------------------
     private async initAsync() {
+        let {api, pendingValue} = this,
+            initialToken;
+
         try {
-            const views = await this.api.fetchViewInfosAsync();
-            runInAction(() => (this.views = views));
+            // Initialize views and related state
+            const views = await api.fetchViewInfosAsync(),
+                state = await api.getStateAsync();
+            runInAction(() => {
+                this.views = views;
+                this.userPinned = state.userPinned;
+                this.autoSave = state.autoSave;
+            });
 
-            if (this.persistWith) {
-                this.initPersist(this.persistWith);
-                await when(() => !this.selectTask.isPending);
-            }
+            // Initialize/choose current view.  Null is ok, and will yield default.
+            initialToken = state.currentView;
+            const initialView =
+                (initialToken ? find(this.views, {token: initialToken}) : null) ??
+                this.initialViewSpec?.(views);
 
-            // If the initial view not initialized from persistence, assign it.
-            if (!this.view) {
-                await this.loadViewAsync(this.initialViewSpec?.(views), this.pendingValue);
-            }
+            await this.loadViewAsync(initialView, pendingValue);
         } catch (e) {
-            // Always ensure at least default view is installed.
-            if (!this.view) this.loadViewAsync(null, this.pendingValue);
-
+            // Always ensure at least default view is installed (other state defaults are fine)
+            this.loadViewAsync(null, pendingValue);
             this.handleException(e, {showAlert: false, logOnServer: true});
         }
 
+        // Add reactions
         this.addReaction({
             track: () => [this.pendingValue, this.autoSave],
             run: () => this.maybeAutoSaveAsync(),
             debounce: 5 * SECONDS
         });
+
+        this.addReaction({
+            track: () => [this.view, this.userPinned, this.autoSave],
+            run: () => api.updateStateAsync(),
+            debounce: 5 * SECONDS,
+            fireImmediately: this.view.token != initialToken
```

Comment:
Curious about the `fireImmediately`. Wondering why not wait until the user has changed something?

Reply:
wanted to catch case where existing view was being re-written because it was no longer present, or a different one.  On the fence here for sure

Reply:
I see- so when a user's current view is no longer available, either the "default" view or a code-specified "initial view" will be loaded, and that view will be immediately persisted as the user's current view?  I wonder if it would be better to clean (remove) their persisted current view in that case since it's no longer a view they chose.  To do that might just adding unnecessary complexity though.

Along similar lines - is it strange that pinning a view will also cause a user's current view to be persisted?  I'm not sure it's really an issue, but wanted to flag.